azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -128,52 +128,21 @@ def __init__(self):
                     return
                 # check whether cgroup monitoring is supported on the current distro
-                self._cgroups_supported = CGroupUtil.cgroups_supported()
+                self._cgroups_supported = self._check_cgroups_supported()
                 if not self._cgroups_supported:
-                    log_cgroup_info("Cgroup monitoring is not supported on {0}".format(get_distro()), send_event=True)
                     # If a distro is not supported, attempt to clean up any existing drop in files in case it was
                     # previously supported. It is necessary to cleanup in this scenario in case the OS hits any bugs on
                     # the kernel related to cgroups.
-                    log_cgroup_info("Agent will reset the quotas in case distro: {0} went from supported to unsupported".format(get_distro()), send_event=False)
-                    self._reset_agent_cgroup_setup()
-                    return
-
-                # check that systemd is detected correctly
-                if not systemd.is_systemd():
-                    log_cgroup_warning("systemd was not detected on {0}".format(get_distro()))
-                    return
-
-                log_cgroup_info("systemd version: {0}".format(systemd.get_version()))
-
-                if not self.__check_no_legacy_cgroups():
-                    return
-
-                # Determine which version of the Cgroup Api should be used. If the correct version can't be determined,
-                # do not enable resource monitoring/enforcement.
-                try:
-                    self._cgroups_api = get_cgroup_api()
-                except InvalidCgroupMountpointException as e:
-                    # Systemd mounts the cgroup file system at '/sys/fs/cgroup'. Previously, the agent supported cgroup
-                    # usage if a user mounted the cgroup filesystem elsewhere. The agent no longer supports that
-                    # scenario. Cleanup any existing drop in files in case the agent previously supported cgroups on
-                    # this machine.
-                    log_cgroup_warning("The agent does not support cgroups if the default systemd mountpoint is not being used: {0}".format(ustr(e)), send_event=True)
-                    log_cgroup_info("Agent will reset the quotas in case cgroup usage went from enabled to disabled")
-                    self._reset_agent_cgroup_setup()
-                    return
-                except CGroupsException as e:
-                    log_cgroup_warning("Unable to determine which cgroup version to use: {0}".format(ustr(e)), send_event=True)
-                    return
-
-                # TODO: Move this and systemd system check to cgroups_supported logic above
-                if self.using_cgroup_v2():
-                    log_cgroup_info("Agent and extensions resource monitoring is not currently supported on cgroup v2")
+                    if not self.using_cgroup_v2():
```

Comment:
We did cleanup when we determine distro is not in supported list. At that point, I doubt we have enabled for v2 machines and later disabled. 

Now we combined several checks for cgroups not supported, so we may enter Line #143 due to other conditions as well. With that in mind, I added only v1. But no issue doing it for every vm. We only touch config if leftover files present, otherwise it's no-op